content/ADR-131-chat-and-friends-protocol.md

```diff
@@ -0,0 +1,105 @@
+
+## API definitions
+
+Worth noting that real time updates like chat messages or friend requests would be streamed, instead of using a request-response calls.
+
+Work in progress: once the APIs are well defined, message definitions should be added.
+
+### Chat API
+
+Currently Kernel is initializing chat & friends when renderer is loaded, this behaviour can be changed, by adding a request for the initialization.
+
+```protobuf
+service ChatAPIService {
+  rpc Initialize(InitializeRequest) returns InitializeResponse
+  rpc SendMessage(SendMessageRequest) returns SendMessageResponse {}
+
+  // streams
+  rpc ReceiveMessages(ReceiveMessageRequest) returns (stream ChatMessage)
+}
+```
+
+### Channels API
+
+```protobuf
+service ChannelsAPIService {
+  // actions
+  rpc CreateChannel(CreateChannelRequest) returns CreateChannelResponse {}
+  rpc JoinOrCreateChannel(JoinOrCreateChannelRequest) returns JoinOrCreateChannelResponse {}
+  rpc LeaveChannel(LeaveChannelRequest) returns LeaveChannelResponse {}
+  rpc MuteChannel(MuteChannelRequest) returns MuteChannelResponse {}
+  rpc MarkChannelMessagesAsSeen(MarkChannelMessagesAsSeenRequest) returns MarkChannelMessagesAsSeenResponse {}
+
+  // queries
+  rpc GetChannels(GetChannelsRequest) returns GetChannelsResponse {}
+  rpc GetJoinedChannels(GetJoinedChannelsRequest) returns GetJoinedChannelsResponse {}
+  rpc GetChannelMessages(GetChannelMessagesRequest) returns (stream GetChannelMessagesResponse {})
+  rpc GetUnseenMessagesByChannel(GetUnseenMessagesByChannelRequest) returns (stream GetUnseenMessagesByChannelResponse {})
+
+  // streams
+  rpc GetChannelInfo(GetChannelInfoRequest) returns (stream GetChannelInfoResponse {})
+  rpc GetChannelMembers(GetChannelMembersRequest) returns (stream GetChannelMembersResponse {})
+}
+
+```
+
+### Friends API
+
+```protobuf
+service FriendsAPIService {
+  // actions
+  rpc MarkMessagesAsSeen(MarkMessagesAsSeenRequest) returns MarkMessagesAsSeenResponse {}
+  rpc UpdateFriendshipStatus(UpdateFriendshipStatusRequest) returns UpdateFriendshipStatusResponse {}
+
+  // queries
+  rpc GetFriends(GetFriendsRequest) returns GetFriendsResponse {}
+  rpc GetFriendsWithDirectMessages(GetFriendsWithDirectMessagesRequest) returns GetFriendsWithDirectMessagesResponse {}
+  rpc GetPrivateMessages(GetPrivateMessagesRequest) returns (stream GetPrivateMessagesResponse {})
+  rpc GetFriendRequests(GetFriendRequestsRequest) returns (stream GetFriendRequestsResponse {})
+}
+
+```
+
+# Benefit
+
+De/serializing JSON can be very slow and waste a lot of time collecting the garbage. Using decentraland protocol and RPC would improve performance and mantainability.
+Also, developers would be talking in the same language and would not have to replicate definitions in both sides, making it reliable and secure.
+In the same direction, these features can be extracted in the future into a service that would not run in the Kernel, simplifying code and improving performance.
+
+# Competition
```

Comment:
Might add: Do nothing and continue using the JSON implementation with the before mentioned issues to performance and maintainability.